components/Typography/src/UIFont+MaterialScalable.m

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "UIFont+MaterialScalable.h"
+
+#import <objc/runtime.h>
+
+#import "MDCTypography.h"
+#import "UIApplication+AppExtensions.h"
+
+static char MDCFontScaleObjectKey;
+
+@implementation UIFont (MaterialTypography)
+
+- (UIFont *)mdc_scaledFontForSizeCategory:(UIContentSizeCategory)sizeCategory {
+  if (!self.mdc_scalingCurve) {
+    return self;
+  }
+
+  NSNumber *fontSizeNumber;
+  if (sizeCategory) {
+    fontSizeNumber = self.mdc_scalingCurve[sizeCategory];
+  }
+
+  // Guard against broken / incomplete scaling curves by returning self if fontSizeNumber is nil.
+  if (fontSizeNumber == nil) {
+    return self;
+  }
+
+  CGFloat fontSize = (CGFloat)fontSizeNumber.doubleValue;
+  UIFont *scaledFont = [UIFont fontWithDescriptor:self.fontDescriptor size:fontSize];
+  scaledFont.mdc_scalingCurve = self.mdc_scalingCurve;
+
+  return scaledFont;
+}
+
+/**
+ @return Device's current UIContentSizeCategory or UIContentSizeCategoryLarge
+     if we are unable to query the device due to being in an extension.
+ */
+static UIContentSizeCategory GetCurrentSizeCategory(void) {
```

Comment:
Tip: Please put this static method at the top of the file.

Reply:
Done.